make/autoconf/flags-cflags.m4

```diff
@@ -929,17 +920,12 @@ AC_DEFUN([FLAGS_SETUP_CFLAGS_CPU_DEP],
 # $2 - Prefix for compiler variables (either BUILD_ or nothing).
 AC_DEFUN([FLAGS_SETUP_GCC6_COMPILER_FLAGS],
 [
-  # These flags are required for GCC 6 builds as undefined behavior in OpenJDK code
-  # runs afoul of the more aggressive versions of these optimizations.
-  # Notably, value range propagation now assumes that the this pointer of C++
-  # member functions is non-null.
-  NO_DELETE_NULL_POINTER_CHECKS_CFLAG="-fno-delete-null-pointer-checks"
-  FLAGS_COMPILER_CHECK_ARGUMENTS(ARGUMENT: [$NO_DELETE_NULL_POINTER_CHECKS_CFLAG],
-      PREFIX: $2, IF_FALSE: [NO_DELETE_NULL_POINTER_CHECKS_CFLAG=""])
+  # This flag is required for GCC 6 builds as undefined behavior in OpenJDK code
+  # runs afoul of the more aggressive versions of this optimization.
   NO_LIFETIME_DSE_CFLAG="-fno-lifetime-dse"
   FLAGS_COMPILER_CHECK_ARGUMENTS(ARGUMENT: [$NO_LIFETIME_DSE_CFLAG],
       PREFIX: $2, IF_FALSE: [NO_LIFETIME_DSE_CFLAG=""])
-  $1_GCC6_CFLAGS="${NO_DELETE_NULL_POINTER_CHECKS_CFLAG} ${NO_LIFETIME_DSE_CFLAG}"
+  $1_GCC6_CFLAGS=${NO_LIFETIME_DSE_CFLAG}
```

Comment:
Might be safer to leave the quote around the DSE flag. Also, why not flag(s) instead?

Reply:
I reverted the quotes.
Didn't get your question; are you suggesting to rename the variable? I'd rather not do that, because 1) it increases the patch size, 2) all variables are global, so renaming might have unintended consequences, and 3) the old name is good enough.

Reply:
Ah no, I meant the comment, from This flag -> These flag(s), since it's implied that this site is where multiple gcc specific flags should be placed in the future. Just a small nit, nothing serious

Reply:
Ah. This place is for GCC6-related flags only, other flags are added in other places. I'm pretty sure we won't add any more flags for GCC6.